### DISCUSSION TITLE
why i cannot use where on join clousure ?

### DISCUSSION THREAD

**Discussion by wznuhidayat** [esyede/rakit#34] [Diskusi Umum]
->left_join('follower as uf2', function($join) {
        $join->on('uf1.follower_id', '=', 'uf2.followed_id')
             ->where('uf2.follower_id', '=', Session::get('user_id'));
    })

**Comment by esyede** [comment]
@wznuhidayat Hi, correct. The join's `on()` method isn't bound to a query class, so you can't chain any other method on those. You can try to put the where clause outside of closure like this:
```php
DB::table('users')->left_join('follower as uf2', function ($join) {
    $join->on('uf1.follower_id', '=', 'uf2.followed_id');
})
->where('uf2.follower_id', '=', Session::get('user_id'))
->select()
->to_sql(); // Use ->get() to retrieve the results
```
Query result:
```sql
SELECT * FROM `users` LEFT JOIN `follower` AS `uf2` ON `uf1`.`follower_id` = `uf2`.`followed_id` WHERE `uf2`.`follower_id` = ?
```

**Reply by wznuhidayat** [reply]
Thank you, I think this is going to work.
